crates/bevy_mesh/src/primitives/dim2.rs

```diff
@@ -452,6 +455,28 @@ pub struct RegularPolygonMeshBuilder {
     circumradius: f32,
     sides: u32,
 }
+
+impl RegularPolygonMeshBuilder {
+    /// Creates a new [`RegularPolygonMeshBuilder`] from the radius of a circumcircle and a number
+    /// of sides.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `circumradius` is negative, or if `sides` is less than 3.
+    pub const fn new(circumradius: f32, sides: u32) -> Self {
+        assert!(
```

Comment:
The asserts should probably be debug asserts, panicking in production is not good and these code paths are unlikely to lead to data corruption.

Reply:
Yup! As a fun note, the mesh builder tries to allocate an incredibly large `Vec` if `sides < 2`.